mesonbuild/dependencies/dev.py

```diff
@@ -404,7 +404,19 @@ class LLVMDependencyCMake(CMakeDependency):
     def __init__(self, name: str, env: 'Environment', kwargs: T.Dict[str, T.Any]) -> None:
         self.llvm_modules = stringlistify(extract_as_list(kwargs, 'modules'))
         self.llvm_opt_modules = stringlistify(extract_as_list(kwargs, 'optional_modules'))
-        super().__init__(name, env, kwargs, language='cpp')
+
+        compilers = None
+        if kwargs.get('native', False):
+            compilers = env.coredata.compilers.build
+        else:
+            compilers = env.coredata.compilers.host
+        if not compilers and not all(x in compilers for x in ('c', 'cpp')):
+            if kwargs.get('required', True):
+                raise self._gen_exception(f'The CMake LLVM dependency needs both a C and C++ compiler')
```

Comment:
Is an exception the right thing to do here? It's possible to fall back on another method, right?

Reply:
@mensinda ping

I think if we just print a warning here (and return early with `self.found = False`), we will anyways error out when the dependency is not found, depending on the value of `required` *and* whether or not other methods were tried and succeeded/failed.